### PR TITLE
Feat: stats detail page

### PR DESCRIPTION
**Pull request by dorianim**
Adds detail page to stats (#4)

Needs: https://github.com/speedclimbing/microservices/pull/39

## Todo
- [x] Unify handling of filter
- [x] Unify handling of table columns
- [x] Make "see more" buttons on season summary page work
- [x] change `AlternativeButton` to be `a` instead of `button`

### CODE REVIEW

**Review by lamorak13 — CHANGES_REQUESTED** [review]
I just checked everything you did .... really impressive. I could follow most of the changes but maybe it would be good to get together one time so that you can update me on what changed. 

Also love the way you handled the stats details page from a UX perspective! :)

File: src/components/shared/buttons/AlternativeButton.svelte [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+	export let onClick: (() => void) | undefined = undefined;
 </script>
 
-<button
-	class={`py-2 w-[150px] bg-grey/5 text-black rounded-[3px] dark:bg-grey text-[14px] dark:text-white ${style}`}
-	on:click={onClick}
-	>{text}
-	<ChevronRight class="inline-block" size="15" /></button
->
+{#if onClick}
```
[the remark this answers is not in the document]
Comment (dorianim) [reply src/components/shared/buttons/AlternativeButton.svelte:9]: Sometimes, we need a button, when we want to execute JavaScript after a click, and sometimes, we need an `a` when we want to navigate.
But you are right, it's not spotless at the moment. Maybe we should use two visually different components for those two use cases?

File: src/components/stats/SmartTable.svelte [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<script lang="ts">
```
[the remark this answers is not in the document]
Comment (dorianim) [reply src/components/stats/SmartTable.svelte:1]: Yes, exactly. The name might be a bit confusing, if you have a better idea, I'm happy to change it :)

**Review by lamorak13 — APPROVED** [review]
LGTM!

**Pull request merged**